boot/bootutil/src/bootutil_priv.h

```diff
@@ -238,14 +259,34 @@ int boot_read_swap_state(const struct flash_area *fap,
 int boot_read_swap_state_by_id(int flash_area_id,
                                struct boot_swap_state *state);
 int boot_write_magic(const struct flash_area *fap);
-int boot_write_status(struct boot_loader_state *state, struct boot_status *bs);
-int boot_schedule_test_swap(void);
+int boot_write_status(const struct boot_loader_state *state, struct boot_status *bs);
 int boot_write_copy_done(const struct flash_area *fap);
 int boot_write_image_ok(const struct flash_area *fap);
 int boot_write_swap_info(const struct flash_area *fap, uint8_t swap_type,
                          uint8_t image_num);
 int boot_write_swap_size(const struct flash_area *fap, uint32_t swap_size);
 int boot_read_swap_size(int image_index, uint32_t *swap_size);
+int boot_erase_trailer_sectors(const struct boot_loader_state *state,
```

Comment:
Personally, I'd like the new swap implementations to have their own (shared) header file. To make it even more decoupled.

Reply:
Done with 5887060ac0f7636793b1a173693c9facb0e6e302. I did not go too far, simply moved new stuff from the PR, but it looks like it would be possible to move more non `overwrite-only` stuff in there... would rather do later...